### ISSUE TITLE
CA1822 false positive for first switch expression when inside expression bodied function

### ISSUE THREAD

**Issue by aboondeviant** [dotnet/roslyn-analyzers#2925]
#### Analyzer package

[Microsoft.CodeAnalysis.FxCopAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.FxCopAnalyzers)

#### Package Version

v2.9.6 (Latest)

#### Diagnostic ID

Example: [CA1822](https://docs.microsoft.com/en-us/visualstudio/code-quality/ca1822-mark-members-as-static)

#### Repro steps
If a switch expression uses a non-static value, combined with an expression bodied function it will sometimes generate a false positive on CA1822.
This only happens if it's the very first switch expression inside the file.

```csharp
public class Test
{
    private string _someValue = "SomeValue";
    private string _someOtherValue = "SomeOtherValue";

    // CA1822 triggers on this function
    public string ExpressionBody1(int a) =>
        a switch
        {
            _ => _someValue
        };
    
    // CA1822 doesn't trigger on this function
    public string ExpressionBody2(int a) =>
        a switch
        {
            _ => _someOtherValue
        };
}
```

#### Expected behavior
No CA1822 on any expression bodied functions with a switch expressions that use a non-static values.

#### Actual behavior
CA1822 is generated

**Comment by goyzhang** [comment]
I'm not able to reproduce this even in a new .cs file only with the sample code you provide.

**Comment by aboondeviant** [comment]
I'm sorry, I've also checked with a colleague in Visual Studio and am unable reproduce it as well.
And this only seems to be happening inside the Rider IDE, I will file a ticket with them.
